spec/unit/provider/service/windows_spec.rb

```diff
@@ -151,33 +120,25 @@
 
   describe "#enabled?" do
     it "should report a service with a startup type of manual as manual" do
-      config.start_type = Win32::Service.get_start_type(Win32::Service::SERVICE_DEMAND_START)
-
+      service_util.expects(:service_start_type).with(name).returns('SERVICE_DEMAND_START')
       expect(provider.enabled?).to eq(:manual)
     end
 
     it "should report a service with a startup type of disabled as false" do
-      config.start_type = Win32::Service.get_start_type(Win32::Service::SERVICE_DISABLED)
-
+      service_util.expects(:service_start_type).with(name).returns('SERVICE_DISABLED')
       expect(provider.enabled?).to eq(:false)
     end
 
-    it "raises an error if the service doesn't exist" do
-      Win32::Service.expects(:config_info).with(name).raises(SystemCallError, 'OpenService')
-
-      expect {
-        provider.enabled?
-      }.to raise_error(Puppet::Error, /Cannot get start type for #{name}/)
-    end
-
     # We need to guard this section explicitly since rspec will always
     # construct all examples, even if it isn't going to run them.
     if Puppet.features.microsoft_windows?
```

Comment:
I don't think this check is necessary anymore.

Reply:
I don't understand why, but without this guard rspec will try to run these examples outside windows because they are defined in a loop.

(I actually had to add this guard to some new tests too)